velox/connectors/hive/iceberg/tests/IcebergReadTest.cpp

```diff
@@ -335,8 +336,9 @@ class HiveIcebergTest : public HiveConnectorTestBase {
     return vectors;
   }
 
-  std::shared_ptr<ConnectorSplit> makeIcebergSplit(
+  std::vector<std::shared_ptr<ConnectorSplit>> makeIcebergSplit(
       const std::string& dataFilePath,
+      const uint32_t splitCount = 1,
```

Comment:
While it's nice to have an option to create multiple splits for one base file, is it necessary to reproduce the bug? We can just construct the delete files and batch sizes such that the failure condition is met. The way you're adding support for multiple splits for one base file is nice, but it needs to be more definitive and easier to calculate the boundaries. Providing just the splitCount is not as straightforward to control the split sizes and the position boundaries. So how about
1. Try to construct a test case using the existing test functions. You may need to add a parameter to control the batch size
2. Add the "multiple splits for one base file" as a new improvement in a separate commit or PR, and instead of just passing in splitCount, pass in a vector of split sizes as the parameter. This way it'll be easier to construct tests with specified split sizes.

Reply:
We are able to reproduce the issues when we run the same tests but with single file multiple splits, all current tests have one file as one split. I have modified the tests to run with 1,2 and 3 splits which reproduce both issues.